packages/gasket-cli/src/commands/create.js

```diff
@@ -61,7 +62,17 @@ class CreateCommand extends Command {
         await loadPreset(context);
         cliVersion(context);
         applyPresetConfig(context);
-        await globalPrompts(context);
+        if (ciConfig) {
+          const parsedCiConfig = JSON.parse(ciConfig);
+          Object.assign(context, parsedCiConfig);
+          await readConfig(context);
+        } else if (ciConfigFilePath) {
+          const ciConfigFile = require(ciConfigFilePath);
+          Object.assign(context, ciConfigFile);
+          await readConfig(context);
+        } else {
+          await globalPrompts(context);
+        }
```

Comment:
See my other comments about still using `globalPrompts` with provide config. Introduce a new `--no-prompts` flags to purposely disable prompting. You can check if that in the `globalPrompts` handlers.

Further, there are still lifecycle driven prompts, which will come from the `promptHooks` action below. I think the simplest approach, would be to continue to execute that lifecycle, but maybe the `prompt` function a noop when `--no-prompt` as been set.

Looking at [this line](https://github.com/godaddy/gasket/blob/pfx-109/packages/gasket-cli/src/scaffold/actions/prompt-hooks.js#L64), you could probably conditional set it to a dummy function that returns an empty object if prompts are disabled. It'd be up to the user to provide the necessary `--config` if also using `--no-prompt`.

Reply:
You could probably tune up `inquirer.prompt` in the globalPrompts, to be have the same way; simple noop if prompts are disabled.